content/post/azure-monitor-terraform.md

```diff
@@ -0,0 +1,230 @@
+      file_patterns = ["/var/log/example/*.log"]
+      settings {
+        text {
+          record_start_timestamp_format = "ISO 8601"
+        }
+      }
+    }
+  }
+  destinations {
+    log_analytics {
+      name                  = local.log_analytics_destination
+      workspace_resource_id = azurerm_log_analytics_workspace.example.id
+    }
+  }
+  data_flow {
+    streams       = ["Custom-${azapi_resource.data_collection_logs_table.name}"]
+    destinations  = [local.log_analytics_destination]
+    output_stream = "Custom-${azapi_resource.data_collection_logs_table.name}"
+  }
+  stream_declaration {
+    stream_name = "Custom-${azapi_resource.data_collection_logs_table.name}"
+    column {
+      name = "TimeGenerated"
+      type = "datetime"
+    }
+    column {
+      name = "Level"
+      type = "string"
+    }
+    column {
+      name = "Logger"
+      type = "string"
+    }
+    column {
+      name = "Context"
+      type = "string"
+    }
+    column {
+      name = "Message"
+      type = "string"
+    }
+    column {
+      name = "AdditionalContext"
+      type = "string"
+    }
+  }
+}
+```
+
+This HCL block will setup the Data Collection Rule. The DCR stream declaration should mirror the schema used the AzAPI resource block used for creating the Log Analytics Workspace Table. The final resources that your configuration should have is:
+
+* Log Analytics Workspace
+* Data Collection Endpoint
+* Azure Monitor Agent Extension
+* Data Collection Rule Association
+
+
+```hcl
+resource "azurerm_log_analytics_workspace" "example" {
+  name                = "example"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+  sku                 = "PerGB2018"
+  retention_in_days   = 30
+}
+
+resource "azurerm_monitor_data_collection_endpoint" "example" {
+  name                          = "mdce"
+  resource_group_name           = azurerm_resource_group.example.name
+  location                      = azurerm_resource_group.example.location
+  kind                          = "Linux"
+}
+
+resource "azurerm_virtual_machine_extension" "ama" {
+  name                       = "AzureMonitorLinuxAgent"
+  virtual_machine_id         = azurerm_virtual_machine.example.id
+  publisher                  = "Microsoft.Azure.Monitor"
+  type                       = "AzureMonitorLinuxAgent"
+  type_handler_version       = "1.0"
+}
+
+resource "azurerm_monitor_data_collection_rule_association" "dcr" {
+  name                    = "dcr"
+  target_resource_id      = azurerm_virtual_machine.example.id
+  data_collection_rule_id = azurerm_monitor_data_collection_rule.example.id
+}
+```
+
+Once you’ve applied out your terraform configuration you can view the table via your Log Analytics Workspace.
+ 
+As stated before, this problem should be fixed in a future version of the azurerm provider but for now I hope it helps someone out!
```

Comment:
The article doesn't link to the documentation of the various things it touches (Log Analytics Workspace, Azure Monitor, Azure Monitor Agent, Terraform…).

The article will be rendered as a web page, not a block of unformatted text. More attention to proper markdown formatting should be given IMHO.